src/serde_helpers.rs

```diff
@@ -380,6 +254,285 @@ pub mod datetime {
+        ///
+        /// The `u32` should represent seconds since the Unix epoch.
+        ///
+        /// Serialization errors if the Timestamp has a non-zero increment.
+        /// ```rust
+        /// # #[cfg(feature = "serde_with-3")]
+        /// # {
+        /// use bson::{serde_helpers::u32, Timestamp};
+        /// use serde::{Serialize, Deserialize};
+        /// use serde_with::serde_as;
+        /// #[serde_as]
+        /// #[derive(Serialize, Deserialize)]
+        /// struct Item {
+        ///     #[serde_as(as = "u32::FromTimestamp")]
+        ///     pub timestamp: Timestamp,
+        /// }
+        /// # }
+        /// ```
+        pub FromTimestamp,
+        Timestamp,
+        |timestamp: &Timestamp| -> Result<u32, String> {
+            if timestamp.increment != 0 {
+                return Err(format!("Cannot convert Timestamp with a non-zero increment to u32: {:?}", timestamp));
+            }
+            Ok(timestamp.time)
+        },
+        |value: u32| -> Result<Timestamp, String> {
+            Ok(Timestamp { time: value, increment: 0 })
+        }
+    );
+
+    serde_conv_doc!(
+        /// Converts a `u32` to and from a [`Timestamp`].
+        ///
+        /// The `u32` should represent seconds since the Unix epoch.
+        ///
+        /// Deserialization errors if the Timestamp has a non-zero increment.
+        /// ```rust
+        /// # #[cfg(feature = "serde_with-3")]
+        /// # {
+        /// use bson::serde_helpers::u32;
+        /// use serde::{Serialize, Deserialize};
+        /// use serde_with::serde_as;
+        /// #[serde_as]
+        /// #[derive(Serialize, Deserialize)]
+        /// struct Event {
+        ///     #[serde_as(as = "u32::AsTimestamp")]
+        ///     pub time: u32,
+        /// }
+        /// # }
+        /// ```
+        pub AsTimestamp,
+        u32,
+        |value: &u32| -> Result<Timestamp, String> {
+            Ok(Timestamp { time: *value, increment: 0 })
+        },
+        |timestamp: Timestamp| -> Result<u32, String> {
+            if timestamp.increment != 0 {
+                return Err(format!("Cannot convert Timestamp with a non-zero increment to u32: {:?}", timestamp));
+            }
+            Ok(timestamp.time)
+        }
+    );
+
+    serde_conv_doc!(
+        /// Converts a `u32` to and from an `f64`.
+        ///
+        /// Deserialization errors if an exact conversion is not possible.
+        /// ```rust
+        /// # #[cfg(feature = "serde_with-3")]
+        /// # {
+        /// use bson::serde_helpers::u32;
+        /// use serde::{Serialize, Deserialize};
+        /// use serde_with::serde_as;
+        /// #[serde_as]
+        /// #[derive(Serialize, Deserialize)]
+        /// struct FileInfo {
+        ///     #[serde_as(as = "u32::AsF64")]
+        ///     pub size_bytes: u32,
+        /// }
+        /// # }
+        /// ```
+        pub AsF64,
+        u32,
+        |value: &u32| -> Result<f64, String> {
+            Ok(*value as f64)
```

Comment:
```suggestion
            Ok(f64::from(*value))
```

it's generally better practice to use a conversion method like `from` or `try_from` rather than `as` when possible. We'll still have to use `as` for the conversion below, though, since there's no `TryFrom<u32>` for `f64`.